atc/db/volume_test.go

```diff
@@ -374,17 +369,76 @@ var _ = Describe("Volume", func() {
 					}),
 				)
 
-				streamedVolume := volumeOnWorker(scenario.Workers[1])
-				err = streamedVolume.InitializeStreamedResourceCache(resourceCache, scenario.Workers[0].Name())
+				scenario.Run(
+					builder.WithBaseWorker(), // workers[1]
+				)
+
+				streamedVolume = volumeOnWorker(scenario.Workers[1])
+			})
+
+			It("InitializeStreamedResourceCache should not fail", func() {
+				err := streamedVolume.InitializeStreamedResourceCache(resourceCache, scenario.Workers[0].Name())
 				Expect(err).ToNot(HaveOccurred())
+			})
 
+			It("should not found the resource cache because source is invalidated", func() {
 				workers, err := defaultTeam.FindWorkersForResourceCache(resourceCache.ID())
 				Expect(err).ToNot(HaveOccurred())
 				Expect(len(workers)).To(Equal(0))
+			})
 
+			It("leaves the volume owned by the container", func() {
 				Expect(streamedVolume.Type()).To(Equal(db.VolumeTypeContainer))
 			})
 		})
+
+		Context("when a streamed resource cache is invalidated", func() {
+			var streamedVolumeOnWorker1 db.CreatedVolume
+			var newVolume db.CreatedVolume
+
+			BeforeEach(func() {
+				scenario.Run(
+					builder.WithBaseWorker(), // workers[1]
+				)
+
+				streamedVolumeOnWorker1 = volumeOnWorker(scenario.Workers[1])
+				err := streamedVolumeOnWorker1.InitializeStreamedResourceCache(resourceCache, scenario.Workers[0].Name())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(streamedVolumeOnWorker1.Type()).To(Equal(db.VolumeTypeResource))
+
+				workers, err := defaultTeam.FindWorkersForResourceCache(resourceCache.ID())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(workers)).To(Equal(2))
+
+				err = scenario.Workers[0].Land()
+				Expect(err).ToNot(HaveOccurred())
+				err = scenario.Workers[0].Prune()
+				Expect(err).ToNot(HaveOccurred())
+
+				foundVolume, found, err := volumeRepository.FindResourceCacheVolume(scenario.Workers[1].Name(), resourceCache)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).To(BeTrue())
+				Expect(foundVolume.Handle()).To(Equal(streamedVolumeOnWorker1.Handle()))
+
+				newVolume = volumeOnWorker(scenario.Workers[1])
+			})
+
+			It("initializing resource cache on the new volume should not fail", func() {
+				err := newVolume.InitializeResourceCache(resourceCache)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("leaves the volume owned by the container", func() {
+				Expect(newVolume.Type()).To(Equal(db.VolumeTypeContainer))
+			})
+
+			It("original invalidated resource cache volume still be found", func() {
```

Comment:
Isn't this one the repeat of line line 418 - 421?

maybe you want to remove those in the before block?
```suggestion
			It("resource cache volume still be found on worker 1 after original invalidated on worker 0", func() {
```